src/visitors/CyclomaticComplexityVisitor.java

```diff
@@ -0,0 +1,36 @@
+package visitors;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.stmt.IfStmt;
+import com.github.javaparser.ast.visitor.GenericVisitorAdapter;
+
+import java.util.List;
+
+public class CyclomaticComplexityVisitor extends GenericVisitorAdapter<Integer, Void> {
+
+    @Override
+    public Integer visit(ClassOrInterfaceDeclaration n, Void arg) {
+        return countDecisions(n);
+
+    }
+
+    private Integer countDecisions(Node node) {
+        Integer counter = 0;
+
+        // count all if statements
+        List<IfStmt> ifStatements = node.findAll(IfStmt.class);
+        counter += ifStatements.size();
```

Comment:
Again, maybe ignore the constructor.

Reply:
I asked Marc what his take is on this.

Reply:
I thought about this and I believe a decision within the constructor should still be counted. If Marc says otherwise I will update this tomorrow.

Reply:
Just remember to mention it in the report.